streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -158,13 +158,19 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
 
     @Override
     public void print() {
-        print(null, null);
+        print(null, null, null);
     }
 
     @Override
-    public void print(Serde<K> keySerde, Serde<V> valSerde) {
+    public void print(String streamName) {
+        print(null, null, streamName);
+    }
+
+    @Override
+    public void print(Serde<K> keySerde, Serde<V> valSerde, String streamName) {
         String name = topology.newName(PRINTING_NAME);
-        topology.addProcessor(name, new KeyValuePrinter<>(keySerde, valSerde), this.name);
+        streamName = (streamName == null) ? this.name : streamName;
```

Comment:
Hmm, I thought you planned to use the "parent processor node name" as the prefix if it is not specified, which seems more user friendly to me?